pyuvdata/uvdata/uvdata.py

```diff
@@ -4575,7 +4954,11 @@ def _select_metadata(
 
         if freq_inds is not None:
             self.Nfreqs = len(freq_inds)
-            self.freq_array = self.freq_array[:, freq_inds]
+            if self.future_array_shapes:
+                self.freq_array = self.freq_array[freq_inds]
+                self.channel_width = self.channel_width[freq_inds]
```

Comment:
I think this is another place that needs handling for `flex_spw` (with respect to `channel_width`) -- I must have missed adding handling for this in the original PR.

Reply:
Yes, thanks for finding that! I'm bothered that tests passed without that. Means I'm probably missing a test that I need to add.